lib/jobs/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2017, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    urlParse = require('url-parse');
+
+module.exports = UcsServiceProfileDiscoveryJobFactory;
+di.annotate(UcsServiceProfileDiscoveryJobFactory, new di.Provide('Job.Ucs.Service.Profile.Discovery'));
+di.annotate(UcsServiceProfileDiscoveryJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Encryption',
+    '_',
+    'JobUtils.UcsTool',
+    'Constants'
+));
+
+function UcsServiceProfileDiscoveryJobFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    encryption,
+    _,
+    UcsTool,
+    Constants
+) {
+    var logger = Logger.initialize(UcsServiceProfileDiscoveryJobFactory);
+
+    /**
+     * @param {Object} options task options object
+     * @param {Object} context graph context object
+     * @param {String} taskId running task identifier
+     * @constructor
+     */
+    function UcsServiceProfileDiscoveryJob(options, context, taskId) {
+        UcsServiceProfileDiscoveryJob.super_.call(this,
+                                   logger,
+                                   options,
+                                   context,
+                                   taskId);
+
+        assert.object(this.options);
+        assert.string(this.options.uri);
+        assert.string(this.options.username);
+        assert.string(this.options.password);
+        assert.string(this.options.ucs);
+        var parse = urlParse(this.options.uri);
+        var protocol = parse.protocol.replace(':','').trim();
+        this.settings = {
+            uri: parse.href,
+            host: parse.host.split(':')[0],
+            root: parse.pathname,
+            port: parse.port,
+            protocol: protocol,
+            username: this.options.username,
+            password: this.options.password,
+            ucs: this.options.ucs,
```

Comment:
Could you please teach me what's the difference between the setting of `ucs` and `host`? Because I see you use the `ucs` as the host in the query but rather than `host`
```JavaScript
var host = this.settings.ucs;
var url= "/login?host=" + host+ "&user="+ username + "&password=" + password;
```

My understanding is:
 - `host` is the ucs-service address, RackHD use the ucs-service to talk with UCS servers.
 - `ucs` is the target UCS server, the ucs-service proxies the request to UCS server.

Is correct?